src/main/src/test/java/org/geoserver/test/GeoServerSystemTestSupport.java

```diff
@@ -164,9 +165,33 @@ protected SystemTestData createTestData() throws Exception {
     /**
      * credentials for mock requests
      */
-    protected String username, password; 
+    protected String username, password;
+
+    /**
+     * Cached dispatcher, it has its own app context, so it's expensive to build
+     */
+    protected static DispatcherServlet dispatcher;
```

Comment:
Building a dispatcher servlet for every mock request we do is rather expensive, do it once and cache the result (a dispatcher servlet has its own local app context to initialize, even if for us it's empty, creating it is costly)